### DISCUSSION TITLE
How to determine the built-in ETCD version and how to upgrade it

### DISCUSSION THREAD

**Discussion by Heyz6** [k3s-io/k3s#5930] [Q&A]
Hello K3S developers

INSTALL_K3S_SKIP_DOWNLOAD=true INSTALL_K3S_EXEC='server --docker --cluster-init --data-dir /home/k3s/data --disable servicelb,traefik,local-storage,metrics-server --etcd-snapshot-dir /opt/etcd' K3S_TOKEN=SECRET K3S_NODE_NAME='node1' /root/k3s-install.sh

If I use an offline K3S deployment scenario, how do I know what version of the built-in ETCD I have installed, and how can I upgrade the version of ETCD if I find a bug, without affecting the cluster operation

**Comment by brandond** [comment]
The embedded etcd version used in each release of k3s is listed in the release notes, and also printed to the logs during startup. You could also query it with etcdctl.

Since it is built into the k3s binary, it cannot be upgraded by any method other than upgrading k3s itself.

**Reply by Heyz6** [reply]
Will it affect my business if I upgrade K3S? systemctl restart k3s. @brosner

**Reply by brandond** [reply]
The workload pods will continue running, but the apiserver will be unavailable while k3s is restarting. If packaged components like coredns or traefik have been upgraded in the new version, you may see short additional outages as those pods are restarted. 

If you're concerned about it, test the upgrade in a non-production environment first.
